### ISSUE TITLE
4.13 bug: class called `row` within module crashes the typechecker

### ISSUE THREAD

**Issue by nojb** [ocaml/ocaml#10661]
```ocaml
module M = struct
  class row = object
  end
end
```
gives
```bash
nojebar@PERVERSESHEAF:~/ocaml$ local/bin/ocamlc -c gui.ml
Fatal error: exception File "typing/signature_group.ml", line 52, characters 19-25: Assertion failed
```
The error goes away if the class is renamed to something other than `row`.

**Comment by gasche** [comment]
Great job finding a bug the day *after* the release :-D

**Comment by gasche** [comment]
Weirdly enough, this example behaves differently depending on whether the repro case is passed as a relative or absolute path.

```
$ ROOT=../../.. ; $ROOT/runtime/ocamlrun $ROOT/ocamlc -nostdlib -I $ROOT/stdlib -c ./pr10661_bad.ml
Fatal error: exception File "typing/signature_group.ml", line 52, characters 19-25: Assertion failed

$ ROOT=../../.. ; $ROOT/runtime/ocamlrun $ROOT/ocamlc -nostdlib -I $ROOT/stdlib -c $(pwd)/pr10661_bad.ml
File "/home/gasche/Prog/ocaml/github-4.13/testsuite/tests/typing-modules-bugs/pr10661_bad.ml", line 1:
Error: The implementation /home/gasche/Prog/ocaml/github-4.13/testsuite/tests/typing-modules-bugs/pr10661_bad.ml
       does not match the interface (inferred signature):  ... In module M:
       The type `#row' is required but not provided
       File "/home/gasche/Prog/ocaml/github-4.13/testsuite/tests/typing-modules-bugs/pr10661_bad.ml", lines 9-10, characters 2-5:
         Expected declaration
```

**Comment by xavierleroy** [comment]
If you guys wanted to ruin the week-end of the release manager, you've succeeded 100%.

**Comment by gasche** [comment]
I proposed a fix in #10662.

**Comment by Octachron** [comment]
I am not sure about the week-end, but it does seem that I will have to cut a 4.13.1 release in the upcoming week.

**Comment by Octachron** [comment]
Fixed by #10662, I will release 4.13.1 with this fix by the end of the week.
